src/modules/app/route-configs/RouteConfigInterface.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+
+import type { Location, Route } from 'redux-first-router'
+import {
+  CityModel,
+  Payload,
+  ExtraModel,
+  PoiModel,
+  CategoriesMapModel,
+  SprungbrettModel,
+  WohnenOfferModel,
+  PageModel,
+  EventModel
+} from '@integreat-app/integreat-api-client'
+import type { TFunction } from 'react-i18next'
+
+export type AllPayloadsType = {|
+  citiesPayload: Payload<Array<CityModel>>,
+  categoriesPayload: Payload<CategoriesMapModel>,
+  poisPayload: Payload<Array<PoiModel>>,
+  eventsPayload: Payload<Array<EventModel>>,
+  extrasPayload: Payload<Array<ExtraModel>>,
+  sprungbrettJobsPayload: Payload<Array<SprungbrettModel>>,
+  wohnenPayload: Payload<Array<WohnenOfferModel>>,
+  disclaimerPayload: Payload<PageModel>
+|}
+
+export type GetLanguageChangePathParamsType = {|location: Location, events: ?Array<EventModel>,
+  categories: ?CategoriesMapModel, pois: ?Array<PoiModel>, language: string|}
+
+export type GetPageTitleParamsType = {|t: TFunction, cityName: string, pathname: string, events: ?Array<EventModel>,
+  categories: ?CategoriesMapModel, pois: ?Array<PoiModel>, extras: ?Array<ExtraModel>, offers: ?Array<WohnenOfferModel>,
+  offerHash: ?string
+|}
+
+export interface RouteConfigInterface<T, P> {
+  name: string,
+  route: Route,
+  getRoutePath: T => string,
+  getLanguageChangePath: GetLanguageChangePathParamsType => string | null,
+  getPageTitle: GetPageTitleParamsType => string,
```

Comment:
Suggestion:

(1) Refactor this to:
```js
getPageTitle: {| t: TFunction, location: Location, payloads: P |}
```

Reply:
At least the name of the function is wrong. It does not return the title but a function.

Reply:
It does return a string

Reply:
Oh yeah sorry :man_facepalming:

Reply:
I like the suggestion, but I would also do this in a separate task.